core/domain/exp_services.py

```diff
@@ -369,7 +369,9 @@ def apply_change_list(exploration_id, change_list):
                 elif (
                         change.property_name ==
                         exp_domain.STATE_PROPERTY_INTERACTION_HINTS):
-                    state.update_interaction_hints(change.new_value)
+                    new_hints_list = [state_domain.Hint.from_dict(hint_dict)
```

Comment:
Maybe add a check that what you receive here is actually a list? This check is there in `update_interaction_hints`, but since this step has now moved here, I think that check should be there here as well.

Reply:
Done